craft_providers/bases/centos.py

```diff
@@ -0,0 +1,950 @@
+            ) from error
+
+    def _install_snaps(self, *, executor: Executor, deadline: Optional[float]) -> None:
+        """Install snaps.
+
+        Snaps will either be installed from the store or injected from the host.
+        - If channel is `None` on a linux system, the host snap is injected
+          into the provider.
+        - If channel is `None` on a non-linux system, an error is raised
+          because host injection is not supported on non-linux systems.
+
+        :param executor: Executor for target container.
+        :param deadline: Optional time.time() deadline.
+        :raises BaseConfigurationError: if the snap cannot be installed
+        """
+        if not self.snaps:
+            logger.debug("No snaps to install.")
+            return
+
+        for snap in self.snaps:
+            self._check_deadline(deadline)
+            logger.debug(
+                "Installing snap %r with channel=%r and classic=%r",
+                snap.name,
+                snap.channel,
+                snap.classic,
+            )
+
+            # don't inject snaps on non-linux hosts
+            if sys.platform != "linux" and not snap.channel:
+                raise BaseConfigurationError(
+                    brief=(
+                        f"cannot inject snap {snap.name!r} from host on "
+                        "a non-linux system"
+                    ),
+                    resolution=(
+                        "install the snap from the store by setting the "
+                        "'channel' parameter"
+                    ),
+                )
+
+            if snap.channel:
+                try:
+                    snap_installer.install_from_store(
+                        executor=executor,
+                        snap_name=snap.name,
+                        channel=snap.channel,
+                        classic=snap.classic,
+                    )
+                except snap_installer.SnapInstallationError as error:
+                    raise BaseConfigurationError(
+                        brief=(
+                            f"failed to install snap {snap.name!r} from store"
+                            f" channel {snap.channel!r} in target environment."
+                        )
+                    ) from error
+            else:
+                try:
+                    snap_installer.inject_from_host(
+                        executor=executor,
+                        snap_name=snap.name,
+                        classic=snap.classic,
+                    )
+                except snap_installer.SnapInstallationError as error:
+                    raise BaseConfigurationError(
+                        brief=(
+                            f"failed to inject host's snap {snap.name!r} "
+                            "into target environment."
+                        )
+                    ) from error
+
+    def _setup_os_special_cmd(self, *, executor: Executor) -> None:
```

Comment:
CentOS has `network` enabled by default (At least working in my system). But that network configuration is not compatible with our current (`ubuntu` / `networkd`) configuration, and our all other bases use `networkd`. We can leave it alone, but I don't know if it will cause any issue.

Reply:
If `network` worked by default for you, I vote we take the easiest path: use `network` and don't bother with switching to `systemd-networkd`.

If `network` doesn't work for another developer, then `yum` won't be able to download `systemd-networkd` and the developer will still be stuck with a broken instance.

If we run into network connectivity issues in the future (on launchpad, with proxies enabled, with docker installed, etc), we can explore some options, possibly:
1. Downloading `systemd-networkd` packages on the host and injecting them into the instance
2. Changing the `network` config settings to match `networkd`'s setting on the Ubuntu bases